test/e2e/framework/pods.go

```diff
@@ -90,24 +102,35 @@ func (f *Framework) PodClientNS(namespace string) *PodClient {
 type PodClient struct {
 	f *Framework
 	v1core.PodInterface
+	skipFrames int
 }
 
 // Create creates a new pod according to the framework specifications (don't wait for it to start).
 func (c *PodClient) Create(pod *v1.Pod) *v1.Pod {
+	return c.create(c.skipFrames+1, pod)
+}
+
+func (c *PodClient) create(skipFrames int, pod *v1.Pod) *v1.Pod {
+	skipFrames++
 	c.mungeSpec(pod)
 	p, err := c.PodInterface.Create(context.TODO(), pod, metav1.CreateOptions{})
-	ExpectNoError(err, "Error creating Pod")
+	ExpectNoErrorWithOffset(skipFrames, err, "Error creating pod")
 	return p
 }
 
 // CreateSync creates a new pod according to the framework specifications, and wait for it to start and be running and ready.
 func (c *PodClient) CreateSync(pod *v1.Pod) *v1.Pod {
+	return c.createSync(c.skipFrames+1, pod)
+}
+
+func (c *PodClient) createSync(skipFrames int, pod *v1.Pod) *v1.Pod {
+	skipFrames++
```

Comment:
This `skipFrames` stuff is a bit hard to follow, and seems fragile. What if instead you just iterate through the callers until you get one outside this file? e.g.

```go
func (c *PodClient) callerOffset() int {
  _, podClientFile, _, ok := runtime.Caller(0)
  if !ok {
    return c.skipFrames+2; // callerOffset() + caller
  }
  for i := 2; i < MaxDepth; i++ {
    _, f, _, ok := runtime.Caller(i)
    if f != podClientFile || !ok {
      return c.skipFrames + i
    }
  }
  // call stack too deep
  return c.skipFrames + MaxDepth
}
```

Reply:
That doesn't work when the code in pods.go is called by some other helper function. Then we would log that helper function instead of the test code.

Skipping frames by counting them is a normal pattern for logging and also how Gomega does it. I agree that off-by-one errors are problematic, but I don't see a good alternative. The "testing" package allows helper functions to mark themselves as helper, but that only works in unit tests and for stack unwinding by the standard library.